src/chargepoint/reservation/ReservationManager.cpp

```diff
@@ -293,7 +301,7 @@ bool ReservationManager::handleMessage(const ocpp::messages::CancelReservationRe
     response.status = CancelReservationStatus::Rejected;
     for (const Connector* connector : m_connectors.getConnectors())
     {
-        if ((connector->status == ChargePointStatus::Reserved) && (connector->reservation_id == request.reservationId))
+        if ((connector->reservation_id_tag.length() > 0) && (connector->reservation_id == request.reservationId))
```

Comment:
```suggestion
        if ((!connector->reservation_id_tag.empty()) && (connector->reservation_id == request.reservationId))
```

Apart from the suggestion, why do you think this way of checking for reservation is an improvement over the other method?

Reply:
Hi
In the previous method, the last status of the connector was used to detect the reservation when sending the start transaction, but since the preparing status notification must be sent after the user authentication, we could never detect that the connector was reserved.
Also, your suggestion is very good.

Reply:
When the reservation checking logic is changed like this I think it would make sense to also add a function to the `ChargePoint` to ask if a certain connector is reserved or not.
With the current reservation checking logic we can do it by calling:

`ocpp::types::ChargePointStatus getConnectorStatus(unsigned int connector_id)`. When you decouple whether a connector is reserved or not from the `ChargePointStatus` (which I think makes sense), it would be handy to be able to query this in the implementing application (we do courrently use `getConnectorStatus` for this).